### PR TITLE
infra: keeping system and IDE files in the global gitignore.

### LINKED ISSUE #800: General: Remove IDE specific gitignore
It is the best practice to not include IDE specific directories into the .gitignore file of the repository. 
The gitignore should be independent in any way. Therefore, it should not include any information about the operation system or IDE.
Better: Use a local global .gitignore. Github has also an [article](https://help.github.com/articles/ignoring-files/) about this. 

What do you think about removing everything which is not necessary for the repository?

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR fixed #800 

# Test Plan:
No need.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,43 +1,10 @@
-*.class
 .#*
 
-# Mobile Tools for Java (J2ME)
-.mtj.tmp/
-
 # Package Files #
 *.jar
```
Comment (oopsmonk) [inline .gitignore]: needed?

File: .gitignore [hunk 2]
```diff
@@ -46,21 +13,6 @@ hs_err_pid*
...
-# JIRA plugin
-atlassian-ide-plugin.xml
-
-# Crashlytics plugin (for Android Studio and IntelliJ)
-com_crashlytics_export_strings.xml
-crashlytics.properties
-crashlytics-build.properties
-fabric.properties
-
 buck-out/
```
Comment (oopsmonk) [inline .gitignore]: what are those?

Comment (th0br0) [reply .gitignore]: from pre-bazel, can be removed.

File: .gitignore [hunk 3]
```diff
@@ -76,11 +24,8 @@ pom.xml
...
 
-# Atom conf
-.clang_complete
-
 # Database
 ciri/ciri.db*
 
-# Visual Studio Code conf
+# ignore all VisualStudioCode user settings
 .vscode/
```
Comment (thibault-martinez) [inline .gitignore:8]: What is the reason some are removed and some aren't ?

Comment (oopsmonk) [reply .gitignore:8]: The global .gitignore genarated from `www.gitignore.io` will commit user settings, but I don't think we should keep user those settings. 

```
### VisualStudioCode ###
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json
```

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**